src/implementations/CrunchDB.ts

```diff
@@ -0,0 +1,27 @@
+import { injectable } from 'inversify';
+import { ICrunchDB } from 'crunchDB/interfaces/ICrunchDB';
+import {
+  IIndexedDB,
+  IIndexedDBType,
+} from 'crunchDB/interfaces/business/IIndexedDB';
+
+@injectable()
+export class CrunchDB implements ICrunchDB {
+  private dbService: IIndexedDB;
+
+  constructor(@inject(IIndexedDBType) dbService: IIndexedDB) {
+    this.dbService = dbService;
+    this.initializeDatabase();
```

Comment:
Be sure to create a separate initialize() method for ICrunchDB and do this work there. Async work should never be done in a constructor.